src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -104,16 +107,39 @@ public override void Initialize(AnalysisContext analysisContext)
                         instantiatedTypes.TryAdd(type.BaseType, null);
                     }
 
-                    // Consider class type declared in the CoClass attribute as instantiated
-                    if (coClassAttributeSymbol != null &&
-                        type.TypeKind == TypeKind.Interface &&
-                        type.GetAttributes().FirstOrDefault(x => x.AttributeClass.Equals(coClassAttributeSymbol)) is AttributeData coClassAttribute &&
-                        coClassAttribute.ConstructorArguments.Length == 1 &&
-                        coClassAttribute.ConstructorArguments[0].Kind == TypedConstantKind.Type &&
-                        coClassAttribute.ConstructorArguments[0].Value is INamedTypeSymbol typeSymbol &&
-                        typeSymbol.TypeKind == TypeKind.Class)
+                    // Consider class types declared in the CoClassAttribute or DesignerAttribute as instantiated
+                    if ((coClassAttributeSymbol != null || designerAttributeSymbol != null) &&
```

Comment:
We already had to filter attributes for the `CoClassAttribute` case so I have decided to merge the 2 cases into on single loop. This might be a bit less readable as the loop now mixes 2 behaviors. I couldn't really decided which side was best so please let me know your preference.